2-WebApp-graph-user/2-5-HybridFlow/Pages/Shared/_Layout.cshtml

```diff
@@ -0,0 +1,198 @@
+                ;
+            }
+            else
+            {
+                @Html.Raw($"const graphEndpoint = '';")
+                ;
+            }
+        }
+
+            const headers = new Headers();
+            const bearer = `Bearer ${token}`;
+            headers.append("Authorization", bearer);
+
+            const options = {
+                method: "GET",
+                headers: headers
+            };
+
+            console.log('request made to Graph API at: ' + new Date().toString());
+
+            return fetch(`${graphEndpoint}${path}`, options)
+                .then(async response => {
+                    return await response.json()
+                })
+                .catch(error => console.log(error))
+        }
+
+        async function signOut() {
+            msalInstance.logoutRedirect();
+
+            fetch('/logout', {
+                method: 'POST',
+                mode: 'same-origin',
+                credentials: 'same-origin',
+            });
+        }
+
+
+        (function () {
+            const scopes = ["user.read"];
+        @{
+            var spaCode = Context.Session.GetString("Microsoft.Identity.Hybrid.Authentication");
+            if (!string.IsNullOrEmpty(spaCode))
+            {
+                @Html.Raw($"const code = '{spaCode}';")
+                ;
+                Context.Session.Remove("Microsoft.Identity.Hybrid.Authentication");
+            }
+            else
+            {
+                @Html.Raw($"const code = '';")
+                ;
+            }
+        }
+            if (!!code) {
+                msalInstance
+                    .handleRedirectPromise()
+                    .then(result => {
+                        if (result) {
+                            console.log('MSAL: Returning from login');
+                            return result;
+                        }
+
+                        const timeLabel = "Time for acquireTokenByCode";
+                        console.time(timeLabel);
+                        console.log('MSAL: acquireTokenByCode hybrid parameters present');
+
+                        return msalInstance.acquireTokenByCode({
+                            code,
+                            scopes,
+                        })
+                            .then(result => {
+                                console.timeEnd(timeLabel);
+                                console.log('MSAL: acquireTokenByCode completed successfully', result);
+                            })
+                            .catch(error => {
+                                console.timeEnd(timeLabel);
+                                console.error('MSAL: acquireTokenByCode failed', error);
+                                if (error instanceof msal.InteractionRequiredAuthError) {
+                                    console.log('MSAL: acquireTokenByCode failed, signing out')
+                                    signOut();
```

Comment:
This should sign in, not sign out, e.g. acquireTokenPopup or acquireTokenRedirect